core/controllers/profile.py

```diff
@@ -306,6 +306,17 @@ class SignupPage(base.BaseHandler):
     """The page which prompts for username and acceptance of terms."""
 
     REDIRECT_UNFINISHED_SIGNUPS = False
+    URL_PATH_ARGS_SCHEMAS = {}
+    HANDLER_ARGS_SCHEMAS = {
+        'GET': {
+            'return_url': {
+                'schema': {
+                    'type': 'basestring'
+                },
+                'default_value': None
```

Comment:
The default value here is none because then the validator won't populate it (And would treat it as an optional parameter), the get method can use the alternate value which it currently is using.

Reply:
Yep, makes sense.